### PR TITLE
Add some more lint rules for react

### PR DESCRIPTION
**Pull request by kevinbarabash**
I was trying out some lint rules that we may want to adopt in webapp as well.  I really wanted `"react/default-props-match-prop-types": ["error", { "allowRequiredDefaults": true }]` to work, but there are a number of issues which make it unviable at the moment for our use cases.  See https://github.com/yannickcr/eslint-plugin-react/issues/1593 for details.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 07baeed7b94ec877350081b0c0c17f6024194cc9

https://deploy-preview-435--wonder-blocks.netlify.com

File: packages/wonder-blocks-dropdown/components/action-menu-opener.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ export type SharedProps = {|
     /**
      * Whether the opener is disabled. If disabled, disallows interaction.
      */
-    disabled?: boolean,
+    disabled: boolean,
```
Comment (kevinbarabash) [inline packages/wonder-blocks-dropdown/components/action-menu-opener.js]: `"react/default-props-match-prop-types": ["error", { "allowRequiredDefaults": true }]` found this, but I ended up not including this rule b/c there was a false positive.

File: .eslintrc [hunk 2]
```diff
@@ -51,6 +51,13 @@
         "promise/no-new-statics": "error",
         "promise/no-return-in-finally": "error",
         "react/jsx-handler-names": "error",
+        "react/no-children-prop": "error",
+        "react/no-typos": "error",
+        "react/no-string-refs": "error",
+        "react/no-this-in-sfc": "error",
+        "react/no-unescaped-entities": "error",
+        "react/react-in-jsx-scope": "error",
+        "react/require-render-return": "error",
```
Comment (kumailn) [inline .eslintrc]: think we should also add `react/no-direct-mutation-state`. Also, `react/sort-prop-types` may be helpful

Comment (kevinbarabash) [reply .eslintrc]: `react/no-direct-mutation-state` would be a solid addition.  In the past we tried `react/sort-prop-types` in webapp, but it turned out to be a lot of work for not much benefit.

**Comment by somewhatabstract** [comment]
Looks ok to me.

**Pull request merged**